build/cmake/CCache.cmake

```diff
@@ -12,7 +12,7 @@
 find_program (CCACHE_EXECUTABLE ccache)
 if (CCACHE_EXECUTABLE)
     message (STATUS "Found ccache: ${CCACHE_EXECUTABLE}")
-    option (MONGO_USE_CCACHE "Use CCache when compiling" ON)
+    option (MONGO_USE_CCACHE "Use CCache when compiling" OFF)
```

Comment:
I would prefer that this be specified by a build script or evergreen option, as this pessimises the overwhelmingly common case where CCache will "just work."